medmodels/data_synthesis/mtgan/mtgan.pyi

```diff
@@ -0,0 +1,76 @@
+from pathlib import Path
+from typing import Dict, Optional, Type
+
+from medmodels.data_synthesis.mtgan.builder import MTGANBuilder
+from medmodels.data_synthesis.mtgan.model.gan import (
+    TrainingHyperparameters,
+    TrainingHyperparametersTotal,
+)
+from medmodels.data_synthesis.mtgan.modules.postprocessor import (
+    AttributeType,
+    MTGANPostprocessor,
+    PostprocessingHyperparameters,
+    PostprocessingHyperparametersTotal,
+)
+from medmodels.data_synthesis.mtgan.modules.preprocessor import (
+    MTGANPreprocessor,
+    PreprocessingHyperparameters,
+    PreprocessingHyperparametersTotal,
+)
+from medmodels.data_synthesis.mtgan.mtgan_model import MTGANModel
+from medmodels.data_synthesis.synthesizer import Synthesizer
+from medmodels.medrecord.medrecord import MedRecord
+from medmodels.medrecord.types import Group, MedRecordAttribute
+
+class MTGAN(Synthesizer):
+    seed: int
+
+    _patients_group: Group
+    _concepts_group: Group
+    _time_attribute: MedRecordAttribute
+    _preprocessing_hyperparameters: PreprocessingHyperparametersTotal
+    _postprocessing_hyperparameters: PostprocessingHyperparametersTotal
+    _training_hyperparameters: TrainingHyperparametersTotal
+
+    def __init__(
+        self,
+        preprocessor: Type[MTGANPreprocessor],
+        postprocessor: Type[MTGANPostprocessor],
+    ) -> None: ...
+    @classmethod
+    def builder(cls) -> MTGANBuilder: ...
+    @staticmethod
+    def _set_configuration(
+        mtgan: MTGAN,
+        *,
+        preprocessor: Type[MTGANPreprocessor],
+        postprocessor: Type[MTGANPostprocessor],
```

Comment:
shouldn't it be a different type here? Since technically one could use other preprocessors, postprocessors

Reply:
`nn.Module` possibly